client/question/components/LearnerViewDirective.js

```diff
@@ -894,6 +901,21 @@ tie.directive('learnerView', [function() {
           EventHandlerService.createQuestionStartEvent();
           EventHandlerService.createTaskStartEvent(
             tasks[currentTaskIndex].getId());
+
+          // Only adds intro message if TIE is iframed, meaning the question
+          // is not shown, and there is nothing currently in the feedback
+          // window.
+          if (ParentPageService.isIframed() &&
+            SessionHistoryService.getBindableSessionTranscript().length === 0) {
+            var introMessageIfIframed = [
```

Comment:
Perhaps all this stuff should be part of the addIntroMessageBalloon() method (which could take no args) if it's always the same? No real reason for it to be here, and it's more testable if we put it in the domain object.

In general logic should go in the services/objects and not the controllers.

Reply:
Removed this and added it to SessionHistoryService to be handled there.